src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,89 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permission, Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdAuthStorage]'
+})
+export class AuthStorageDirective {
+  permissions: Permissions;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authStorageService: AuthStorageService
+  ) {
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  @Input('cdAuthStorage') set cdAuthStorage(condition: any) {
+    /*
+      single permission equivalent to check e.g osd.
+      We check if the permission key exists on the Permissions class then
+      if true We check for read permission and display the component accordingly
+    */
+
+    if (_.isString(condition) && condition in this.permissions) {
+      if (this.permissions[condition].read) {
+        this.viewContainer.createEmbeddedView(this.templateRef);
+      } else {
+        this.viewContainer.clear();
+      }
+    }
+
+    /*
+      lodash _isArray matches both objects and arrays.
+      By checking if the first index contains a value we can tell if it is truly an array or not
+      Also works for _isObject
+    */
+
+    if (_.isArray(condition) && condition[0] !== undefined) {
+      const checkedPermissions: boolean[] = [];
+
+      _.forEach(condition, (value) => {
+        if (_.has(this.permissions, [value, 'read'])) {
+          checkedPermissions.push(true);
+        } else {
+          checkedPermissions.push(false);
+        }
+      });
```

Comment:
Looks good! Why not decoupling the create/clear logic from the permission validation:
```ts
@Input('cdAuthStorage') set cdAuthStorage(condition: any) {
  if (this.isAuthorized(condition)) {
    this.viewContainer.createEmbeddedView(this.templateRef);
  } else {
    this.viewContainer.clear();
  }
}

isAuthorized(condition) {
  if (string) {
    ...
    return true|false;
  } else if (array) {
    ...
    return true|false;
  }
}
```